query-engine/connectors/sql-query-connector/src/database/operations/read.rs

```diff
@@ -67,30 +56,24 @@ pub(crate) async fn get_single_record_wo_joins(
     conn: &dyn Queryable,
     model: &Model,
     filter: &Filter,
-    selected_fields: &ModelProjection,
-    aggr_selections: &[RelAggregationSelection],
+    selected_fields: &FieldSelection,
     ctx: &Context<'_>,
 ) -> crate::Result<Option<SingleRecord>> {
+    let selected_fields = selected_fields.without_relations().into_virtuals_last();
+
     let query = read::get_records(
         model,
-        selected_fields.as_columns(ctx).mark_all_selected(),
-        aggr_selections,
+        ModelProjection::from(&selected_fields)
+            .as_columns(ctx)
+            .mark_all_selected(),
```

Comment:
A little out of scope here but generally there's no reason `FieldSelection` couldn't have `as_columns` method itself, that would eliminate `ModelProjection` completely on this code path now that `FieldSelection` and not `ModelProjection` is passed to this function. Related: https://github.com/prisma/team-orm/issues/782

Reply:
Part of the reason why I like that `AsColumns` is only implemented for `ModelProjection` is that it only contains fields that are mappable to a model. Conversely, `FieldSelection` contains fields that aren't, like `Relation` and `Virtual`.

Technically, you are right, we could, but I'm slightly afraid it introduces mismatches in some places.